docs/architecture-and-design.md

```diff
@@ -394,3 +400,62 @@ In order for a robot to move to the desired destination of a `MoveIntent`, the N
 The `Visualizer` is exactly what it sounds like: A visualizion of our [AI](#ai). It provides a GUI that shows us the state of the `World` as the [Backend](#backend) sees it, and is also able to display extra information that the [AI](#ai) would like to show. For example, it can show the planned paths of each friendly robot on the field, or highlight which enemy robots it thinks are a threat. Furthermore, it displays any warnings or status messages from the robots, such as if a robot is low on battery.
 
 The `Visualizer` also lets us control the [AI](#ai) by setting [Dynamic Parameters](#dynamic-parameters). Through the `Visualizer`, we can manually choose what strategy the [AI](#ai) should use, what colour we are playing as (yellow or blue), and tune more granular behaviour such as how close an enemy must be to the ball before we consider them a threat.
+  
+
+# Simulated Integration Tests
+When it comes to gameplay logic, it is very difficult if not impossible to unit test anything higher-level than a [Tactic](#tactics) (and even those can be a bit of a challenge). Therefore if we want to test [Plays](#plays) we need a higher-level integration test that can account for all the indepdent events, sequences of actions, and timings that are not possible to adequately cover in a unit test. For example, testing that a passing play works is effectively impossible to unit test because the logic needed to coordinate a passer and receiver relies on more time-based information like the movement of the ball and robots. We can only validate that decisions at a single point in time are correct, not that the overall objective is achieved successfully.
+
+Ultimately, we want a test suite that validates our [Plays](#plays) are generally doing the right thing. We might not care exactly where a robot receives the ball during a passing play, as long as the pass was successful overall. The solution to this problem is to use simulation to allow us to deterministically run our entire AI pipeline and validate behavior.
+
+The primary goals of this test system are:
+1. **Determinism**
+    * We need tests to pass or fail consistently
+2. **Test "ideal" behaviour**
+    * We want to test the logic in a "perfect world", where we don't care about all the exact limitations of our system in the real world with real physics. Eg. we don't care about modelling robot wheels slipping on the ground as we accelerate.
+
+## Simulated Integration Tests Architecture
+Fortunately, our abstractions and [Observer system](#observer-design-pattern) make it very easy to modify our system to support simulated integration tests. There are only 2 major changes we make to the system:
+1. Implement a [Backend](#backend) to handle the simulation
+2. Add another Observer to the system that will handle the "validation"
```

Comment:
changed as suggested, and added the `AI under test` as a third component to describe the system as a whole, rather than a diff from the "regular" system.

Reply:
Looks like phrasing is unchanged?

Reply:
It is for me?

Reply:
![Screenshot from 2019-12-24 10-58-03](https://user-images.githubusercontent.com/9075711/71423177-46722380-263c-11ea-8835-8c751ad9c8a1.png)

Reply:
Does that look different for you?

Reply:
Go look at the full diff, the previous in the "Conversation" tab isn't updated.

Reply:
That is the full diff, look at the URL?

Reply:
Wait a minute, what you have in your screenshot is the re-phrased version. Whats the issue with the phrasing?

Reply:
Part of my comment above was that we shouldn't write documentation in a "instruction manual" sense. ie. We should say `1. A [Backend[(#backend) that performs the simulation` instead of `1. Implement a [Backend](#backend) that performs the simulation`. Similar comment for `2.`.

Reply:
Oh sorry I was fixing the structure not the phrasing. Should be fixed now.